compaction/task_manager_module.cc

```diff
@@ -297,19 +379,40 @@ static future<bool> maybe_flush_all_tables(sharded<replica::database>& db) {
     co_return false;
 }
 
+future<> global_major_compaction_task_impl::run() {
+    bool flushed_all_tables = false;
+    if (_flush_mode == flush_mode::all_tables) {
+        flushed_all_tables = co_await maybe_flush_all_tables(_db);
+    }
+
+    std::unordered_map<sstring, std::vector<table_info>> tables_by_keyspace;
+    auto tables_meta = _db.local().get_tables_metadata().get_column_families_copy();
+    for (const auto& [table_id, t] : tables_meta) {
+        const auto& ks_name = t->schema()->ks_name();
+        const auto& table_name = t->schema()->cf_name();
+        auto it = tables_by_keyspace.try_emplace(ks_name, std::vector<table_info>()).first;
+        it->second.emplace_back(table_name, table_id);
```

Comment:
`tables_by_keyspce[ks_name].emplace_back(table_name, table_id);`

Reply:
done in next version